### DISCUSSION TITLE
Converting 10 bit YUV DPX file to 10 bit RGB DPX

### DISCUSSION THREAD

**Discussion by kirkkwang** [ImageMagick/ImageMagick#4338] [Help] [answered]
I have around 32,000 DPX files i want to use RAWcooked on to make an MKV file but it doesn't seem to like YUV.  Is there a way to conver these to RGB?

I've been trying
`magick "input.dpx" -set colorspace rgb out.dpx`
which creates an output but it doesn't seem to changing it.  Are there other arguments needed for this?

**Comment by snibgo** [comment]
What version of IM, on what platform?

"-set colorspace rgb" will change metadata but not pixel values. It tells IM that the image is encoded as linear RGB. If it is actually encoded as YUV, the results will be bad.

Please post a sample input.dpx.

**Reply by kirkkwang** [reply]
Oh yes!
ImageMagick 7.0.7-3 Q16 x64 2017-09-18 on Windows 10
[sample.zip](https://github.com/ImageMagick/ImageMagick/files/7313438/sample.zip)

I've attached a sample

**Comment by snibgo** [comment]
Why do you think sample.dpx has YUV channels? According to IM and exiftool, it has only one channel.

**Reply by kirkkwang** [reply]
Sorry, I don't fully understand it, when I ran exiftool on it I'm getting this line:
`Components Configuration        : Luminance (Y)`

On others dpx files that do work with RAWcooked I get
`Components Configuration        : R, G, B`
[sample2.zip](https://github.com/ImageMagick/ImageMagick/files/7313717/sample2.zip)

I guess the better question for me to ask is what is meant by Components Configuration in the exif and how do I change the file to say RGB?

**Comment by snibgo** [comment, marked as answer]
I know nothing about RAWcooked.

Yes, exiftool says your file has only a Y channel, for luminance. Why do you think it also has U and V colour channels?

To convert Y luminance into sRGB luminance, we can do this:
```
magick sample.dpx -set colorspace YUV -channel 1,2 -evaluate set 50% +channel -colorspace sRGB -resize 25% x.jpg
```
I resize and save to JPG for upload speed.
![x](https://user-images.githubusercontent.com/33812368/136617611-314563b0-624f-4102-a197-77e4c7553002.jpg)

There seems to be a problem with the length of each row. I don't know if that is a problem in the file metadata, or if IM is misreading the file.

EDIT: We can correct the apparent row length problem:
```
magick sample.dpx -set colorspace YUV -channel 1,2 -evaluate set 50% +channel -colorspace sRGB -crop x1 +repage +append -crop 2047x +repage -append -resize 25% x2.jpg
```
![x2](https://user-images.githubusercontent.com/33812368/136618643-cb8fce91-4f00-4de5-a394-70fbdb3312b4.jpg)

**Reply by kirkkwang** [reply]
Hey thanks for that correction!  I didn't think about that being a problem in the grand scheme of things but this is obviously better.  I thought since it had a Y channel that meant it would also have a U and a V, clearly I was wrong.

When I ran your code I'm getting this color happening:

![x2](https://user-images.githubusercontent.com/19597776/136621466-2df923f9-d0b6-470c-b035-0eab8c315d7f.jpg)

Am I missing something?

**Comment by snibgo** [comment]
Perhaps you are running it from a BAT script file and haven't doubled the % signs.

Perhaps your rather old version of IM needs updating.

**Reply by kirkkwang** [reply]
Yup, updating fixed it!  Now to do it in a loop....
